osu.Game/Overlays/SocialOverlay.cs

```diff
@@ -166,20 +171,51 @@ private void recreatePanels(PanelDisplayStyle displayStyle)
 
             LoadComponentAsync(newPanels, f =>
             {
-                if (panels != null)
-                    ScrollFlow.Remove(panels);
-
+                loading.Hide();
                 ScrollFlow.Add(panels = newPanels);
             });
         }
 
         private void updateUsers(IEnumerable<User> newUsers)
         {
-            Users = newUsers;
-            loading.Hide();
+            loading.Show();
+
+            var sortDirection = Filter.DisplayStyleControl.Dropdown.Current.Value;
+
+            IEnumerable<User> sortedUsers = newUsers;
+
+            if (sortedUsers.Any())
```

Comment:
can't this just early return?

Reply:
Nope, we need that because `Clear` function call happens in `recreatePanels` which will be called after sorting